delta-lake/src/integration/java/org/apache/iceberg/delta/DeltaLakeToIcebergSparkActions.java

```diff
@@ -24,18 +24,29 @@
 import org.apache.spark.sql.connector.catalog.CatalogPlugin;
 
 /** An example class shows how to use the delta lake migration actions in SparkContext. */
-class DeltaLakeToIcebergMigrationSparkIntegration {
+public class DeltaLakeToIcebergSparkActions implements DeltaLakeToIcebergMigrationActionsProvider {
+  private final SparkSession spark;
 
-  private DeltaLakeToIcebergMigrationSparkIntegration() {}
+  private DeltaLakeToIcebergSparkActions(SparkSession spark) {
+    this.spark = spark;
+  }
+
+  public static DeltaLakeToIcebergSparkActions get(SparkSession spark) {
+    return new DeltaLakeToIcebergSparkActions(spark);
+  }
+
+  public static DeltaLakeToIcebergSparkActions get() {
+    return new DeltaLakeToIcebergSparkActions(SparkSession.active());
+  }
 
-  static SnapshotDeltaLakeTable snapshotDeltaLakeTable(
-      SparkSession spark, String newTableIdentifier, String deltaTableLocation) {
+  @Override
+  public SnapshotDeltaLakeTable snapshotDeltaLakeTable(
+      String tableIdent, String sourceTableLocation) {
     String ctx = "delta lake snapshot target";
     CatalogPlugin defaultCatalog = spark.sessionState().catalogManager().currentCatalog();
     Spark3Util.CatalogAndIdentifier catalogAndIdent =
-        Spark3Util.catalogAndIdentifier(ctx, spark, newTableIdentifier, defaultCatalog);
-    return DeltaLakeToIcebergMigrationActionsProvider.getDefault()
-        .snapshotDeltaLakeTable(deltaTableLocation)
+        Spark3Util.catalogAndIdentifier(ctx, spark, tableIdent, defaultCatalog);
+    return new BaseSnapshotDeltaLakeTableAction(sourceTableLocation)
```

Comment:
Extends from the above. This implementation will be the same if we want to add the action to `iceberg-spark` later

Reply:
Also, I think it is still worth to maintain the `default()` thing in the interface, because it would give users max freedom to integrate these migration actions to their own environment.